### PR TITLE
[BugFix] support several array function for json types

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

- `array_has`, `array_contains`, `array_remove`.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

**Comment by Seaven** [comment]
Add test case?

**Comment by fzhedu** [comment]
please add some tests.

**Review by Seaven — DISMISSED** [review]

**Review by fzhedu — APPROVED** [review]
LGTM

File: test/sql/test_array_fn/R/test_array_fn [hunk 1]
```diff
@@ -1282,6 +1282,29 @@ select array_contains([[1,2,3], [2,3,4]], 100) from array_test order by pk;
...
+select array_contains_all([parse_json('{"addr": 1}'), parse_json('{"addr": 2}')], 
+                          [parse_json('{"addr": 1}'), parse_json('{"addr": 2}')]);
+-- result:
+1
+-- !result
+select array_contains_all([parse_json('{"addr": 1}'), parse_json('{"addr": 2}')], 
+                          [parse_json('{"addr": 1}'), parse_json('{"addr": 2}'), parse_json('{"addr": 3}')]);
+-- result:
+0
+-- !result
```
Comment (fzhedu) [inline test/sql/test_array_fn/R/test_array_fn:1307]: it is better to have some special cases, like null, empty

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Seaven — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**